.github/workflows/update_directory_md.yml

```
@@ -1,59 +1,30 @@
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
-      - uses: actions/setup-python@master
-      - name: update_directory_md
-        shell: python
-        run: |
-            import os
-            from typing import Iterator
-            URL_BASE = "https://github.com/TheAlgorithms/Ruby/blob/master"
-            g_output = []
-            
-            def good_filepaths(top_dir: str = ".") -> Iterator[str]:
-                for dirpath, dirnames, filenames in os.walk(top_dir):
-                    dirnames[:] = [d for d in dirnames if d[0] not in "._"]
-                    for filename in filenames:
-                        if os.path.splitext(filename)[1].lower() == ".rb":
-                            yield os.path.join(dirpath, filename).lstrip("./")
-                            
-            def md_prefix(i):
-                return f"{i * '  '}*" if i else "\n##"
-                
-            def print_path(old_path: str, new_path: str) -> str:
-                global g_output
-                old_parts = old_path.split(os.sep)
-                for i, new_part in enumerate(new_path.split(os.sep)):
-                    if i + 1 > len(old_parts) or old_parts[i] != new_part:
-                        if new_part:
-                            g_output.append(f"{md_prefix(i)} {new_part.replace('_', ' ').title()}")
-                return new_path
-                
-            def build_directory_md(top_dir: str = ".") -> str:
-                global g_output
-                old_path = ""
-                for filepath in sorted(good_filepaths(), key=str.lower):
-                    filepath, filename = os.path.split(filepath)
-                    if filepath != old_path:
-                        old_path = print_path(old_path, filepath)
-                    indent = (filepath.count(os.sep) + 1) if filepath else 0
-                    url = "/".join((URL_BASE, filepath, filename)).replace(" ", "%20")
-                    filename = os.path.splitext(filename.replace("_", " ").title())[0]
-                    g_output.append(f"{md_prefix(indent)} [{filename}]({url})")
-                return "\n".join(g_output)
-            with open("DIRECTORY.md", "w") as out_file:
-                out_file.write(build_directory_md(".") + "\n")
-                
-      - name: Update DIRECTORY.md
-        run: |
-          cat DIRECTORY.md
-          git config --global user.name github-actions
-          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
-          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
-          git add DIRECTORY.md
-          git commit -am "updating DIRECTORY.md" ||  true
-          git push --force origin HEAD:$GITHUB_REF || true
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Build directory
+        uses: TheAlgorithms/scripts/directory_md@main
```

Hey 👋🏼 could one of you create a branch which uses the patch I made in [this PR](https://github.com/TheAlgorithms/scripts/pull/29).

Just change the above line to something like this and it should hopefully work - if it doesn't, ping me directly please and I can take a look at the action log

```suggestion
        uses: TheAlgorithms/scripts/directory_md@e24e3ba9cf7e90c91ac87f953593426328b86563
```

If it works then I can merge my fix and then you won't have to merge this suggested change.